### PR TITLE
Remove placementrule dependency

### PR DESCRIPTION
**Pull request by clyang82**
fixed: https://github.com/open-cluster-management/backlog/issues/14168
Signed-off-by: clyang82 <chuyang@redhat.com>

Merge code to support 311 - https://github.com/open-cluster-management/multicluster-observability-operator/pull/665
https://github.com/open-cluster-management/endpoint-metrics-operator/pull/110

### CODE REVIEW

File: operators/multiclusterobservability/main.go [hunk 1]
```diff
@@ -181,17 +174,12 @@ func main() {
 		workv1.SchemeGroupVersion.WithKind("ManifestWork"): []filteredcache.Selector{
 			{LabelSelector: "owner==multicluster-observability-operator"},
 		},
+		clusterv1.SchemeGroupVersion.WithKind("ManagedCluster"): []filteredcache.Selector{},
```
Comment (morvencao) [inline operators/multiclusterobservability/main.go:177]: why add this if no filter?

File: operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go [hunk 2]
```diff
@@ -408,29 +392,35 @@ func deleteManagedClusterRes(c client.Client, namespace string) error {
...
+	vendor, ok := obj.GetLabels()["vendor"]
+	if !ok && vendor != "OpenShift" {
+		return
+	}
+	obs, ok := obj.GetLabels()["observability"]
+	if ok && obs == "disabled" {
+		return
+	}
+	managedClusterList = util.RemoveDuplicates(append(managedClusterList, obj.GetName()))
+}
```
Comment (morvencao) [inline operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go:405]: I'm thinking can we filter the `managedCluster` resources by the labels in filter cache?

Comment (clyang82) [reply operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go]: Done

**Review by morvencao — COMMENTED** [review]
/lgtm

**Pull request merged**